### DISCUSSION TITLE
Unknown frequencies in electric field for laser-plasma interactions

### DISCUSSION THREAD

**Discussion by JohanKoelsen** [epochpic/epoch#471] [Q&A]
Dear Epoch community,

When running EPOCH simulations with a pump wave, the results seem to produce frequencies in the EM-polarization of the electric field corresponding to the frequency of the pump wave shifted by harmonics of a fixed frequency.

The following picture shows the fourier-transformed EM-polarization of the electric field taken at the middle of the simulation:
![EM_spectrum_opt1](https://user-images.githubusercontent.com/78602003/227505025-1101fc50-c535-4e2b-bf60-91730ba1a20d.png)

The frequency of the laser is 140GHz. From the laser, reoccurring pairs of peaks appear at around every harmonic of 21GHz, which I can't explain. 
I can account for the two peaks at around 70GHz, and the increased values in the spectrum around 64GHz which are due to the EC resonance.

Do any of you know if any numerical artifacts can produce these peaks?

Many thanks,

Johan

The input.deck file is here below (saved as .txt file):

[input.txt](https://github.com/Warwick-Plasma/epoch/files/11061462/input.txt)

**Comment by Status-Mirror** [comment]
This isn't something I know much about personally, but if they are numerical artefacts, then maybe you would expect them to change position as you adjust things like cell size? This would be a good test to see if there any clear signs of a numerical effect.

Cheers,
Stuart

**Reply by JohanKoelsen** [reply]
Thanks for the reply Stuart :) I ran tests with varying number of grid cells, and they did indeed change position. I have scaled down my simulation, and I'm only running a laser-block now:

begin:boundaries
    bc_x_min_field = simple_laser
    bc_x_min_particle = thermal
    bc_x_max_field = simple_outflow
    bc_x_max_particle = thermal

end:boundaries

begin:laser
    boundary = x_min
    id = 1
    intensity_w_cm2 = 1.0e5
    omega = w

    n:constant
    w = 2*pi*140e9
    t_profile = 2/pi*atan(time*5e9)
    pol_angle = pi/2
    t_start = 0.0
end:laser

![Ez_spec](https://user-images.githubusercontent.com/78602003/231387119-26faea10-7788-4393-b63d-60107adb11a5.png)

They do however still appear in the FFT of the Ez field. I will look further into my FFT. But thanks for the tip, I am quite sure now that it is indeed numerical artifacts.

**Reply by amarsarari** [reply]
Dear sir can you provide matlab code for this
